External/FEXCore/Source/Interface/IR/Passes/ConstProp.cpp

```diff
@@ -690,6 +690,23 @@ bool ConstProp::ConstantPropagation(IREmitter *IREmit, const IRListView& Current
       }
     break;
     }
+    case OP_TEST: {
+      auto Op = IROp->CW<IR::IROp_Test>();
+      uint64_t Constant1{};
+      uint64_t Constant2{};
+
+      if (IREmit->IsValueConstant(Op->Header.Args[0], &Constant1) &&
+          IREmit->IsValueConstant(Op->Header.Args[1], &Constant2)) {
+        uint64_t Result = (Constant1 & Constant2) & getMask(Op);
+        bool N = Result & (1ull << ((Op->Header.Size * 8) - 1));
+        bool Z = Result == 0;
+        uint32_t NZVC = (N ? (1 << 31) : 0) | (Z ? (1 << 30) : 0);
```

Comment:
```suggestion
        uint32_t NZVC = (N ? (1U << 31) : 0) | (Z ? (1U << 30) : 0);
```

otherwise some checks in ubsan will complain about this (specifically the `1 << 31` shift)